cluster-service/cspr/namespace-cleaner.yaml

```diff
@@ -0,0 +1,77 @@
+---
+apiVersion: template.openshift.io/v1
+kind: Template
+metadata:
+  name: namespace-cleaner
+parameters:
+  - name: ORPHANED_NAMESPACE_CLEANER_NAMESPACE
+    description: The namespace to create to have a cronjob which will delete the orphaned namespace which are not deleted due to any issues with the jenkins job.
+    value: orphaned-namespace-cleaner
+  - name: NAMESPACE_CLEANER_CLUSTERROLE_NAME
+    value: namespace-cleaner
+  - name: KUBECTL_IMAGE
+    description: An image which have the `kubectl` binary in it.
+    value: quay.io/rhn_support_ansverma/ubi8-minimal-kubectl:latest
+objects:
+  - apiVersion: rbac.authorization.k8s.io/v1
+    kind: ClusterRole
+    metadata:
+      name: ${NAMESPACE_CLEANER_CLUSTERROLE_NAME}
+    rules:
+      - apiGroups:
+          - ""
+        resources:
+          - namespaces
+        verbs: ["get", "list", "delete", "watch"]
+  - apiVersion: v1
+    kind: Namespace
+    metadata:
+      name: ${ORPHANED_NAMESPACE_CLEANER_NAMESPACE}
+  - apiVersion: v1
+    kind: ServiceAccount
+    metadata:
+      name: orphaned-namespace-cleaner-cronjob
+      namespace: ${ORPHANED_NAMESPACE_CLEANER_NAMESPACE}
+  - apiVersion: rbac.authorization.k8s.io/v1
+    kind: ClusterRoleBinding
+    metadata:
+      name: orphaned-namespace-cleaner-cronjob
+    subjects:
+      - kind: ServiceAccount
+        name: orphaned-namespace-cleaner-cronjob
+        namespace: ${ORPHANED_NAMESPACE_CLEANER_NAMESPACE}
+    roleRef:
+      kind: ClusterRole
+      name: ${NAMESPACE_CLEANER_CLUSTERROLE_NAME}
+      apiGroup: rbac.authorization.k8s.io
+  - apiVersion: batch/v1
+    kind: CronJob
+    metadata:
+      name: orphaned-namespace-cleaner
+      namespace: ${ORPHANED_NAMESPACE_CLEANER_NAMESPACE}
+    spec:
+      schedule: "*/1 * * * *"
+      successfulJobsHistoryLimit: 3
+      failedJobsHistoryLimit: 1
+      jobTemplate:
+        spec:
+          template:
+            spec:
+              serviceAccountName: orphaned-namespace-cleaner-cronjob
+              containers:
+                - name: kubectl-container
+                  image: ${KUBECTL_IMAGE}
+                  command: ["/bin/sh", "-c"]
+                  args:
+                    - |
+                      echo "Starting to clear orphaned namespaces"
+                      # `select((now - (.metadata.creationTimestamp | fromdate)) / 60 > 60)` selects the namespaces which are older than 60 minutes.
```

Comment:
The comment doesn't seem to match the executed command

Reply:
Yes, I have only mentioned the part which selects the namespaces based on how old they are so that anyone looking at the definition, understand that it is for the namespaces older than 1 hour(60 mins).

Reply:
Yes but what I meant is that that part doesn't match the command either. One shows /60 > 60 and the other shows /1 > 1

Reply:
Oh, I am sorry, Let me correct it.